src/main/java/com/server/CLIFlagParser.java

```diff
@@ -0,0 +1,27 @@
+package com.server;
+
+import java.io.IOException;
+
+public class CLIFlagParser {
+
+    public ServerConfig parse(String[] args) throws IOException {
```

Comment:
This meets the minimum requirement. Is it this sufficient or is it preferred to do more error handling, or should we wait to hear back from the client/users?